config/testdata/conf.good.yml

```diff
@@ -109,7 +109,8 @@ scrape_configs:
 - job_name: service-kubernetes
 
   kubernetes_sd_configs:
-  - api_servers:
+  - kind: service_endpoints
```

Comment:
> Service endpoints are only useful for blackbox monitoring.
Services are the widely used configuration, which currently have the healthcheck problem.

I think that's more the other way around. Blackbox monitoring should be targeted at services; service endpoints are currently the scraped targets (endpoint == ready pod). See https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L99-L127 for the blackbox monitoring targeting services & https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L57-L91 targeting service endpoints.